src/catalog/src/builtin.rs

```diff
@@ -1894,6 +1894,18 @@ pub static MZ_COMPUTE_DEPENDENCIES: Lazy<BuiltinSource> = Lazy::new(|| BuiltinSo
     is_retained_metrics_object: false,
     sensitivity: DataSensitivity::Public,
 });
+pub static MZ_COMPUTE_HYDRATION_STATUS: Lazy<BuiltinSource> = Lazy::new(|| BuiltinSource {
```

Comment:
I think I've lost track of the differences, but why a `BuiltinSource` over a `BuiltinLog`?

Reply:
A `BuiltinLog` is the same as a replica introspection source, so it's written by replicas and queries are replica-specific. `mz_compute_hydration_status` is not one of these because it's written by the controller and results are not replica-specific. So we have to expose it as a `BuiltinSource`, aka "storage-managed collection", which is a persist collection that's maintained by the storage controller.